Sources/SwiftParser/Types.swift

```diff
@@ -263,8 +263,8 @@ extension Parser {
             )
           )
           break
-        } else if self.atContextualKeyword("Type") || self.atContextualKeyword("Protocol") {
-          let typeOrProtocol = self.consumeAnyToken(remapping: .contextualKeyword)
+        } else if self.at(.contextualKeyword(.Type)) || self.at(.contextualKeyword(.Protocol)) {
+          let typeOrProtocol = self.consume(if: .contextualKeyword(.Type)) ?? self.consume(if: .contextualKeyword(.Protocol))!
```

Comment:
IMO we should add a 2 and 3 arg consume to avoid the array creation and then use that here. The only non-2/3 kind calls should probably be subsets anyway.

Reply:
I agree. I’ll re-consider it after I have had some time to play around automatic generation of `RawTokenKindSubset` from `text_choices` tomorrow.